src/common/Graph.js

```diff
@@ -13,6 +13,27 @@ export const mix = (v1, v2, s) => {
   return new Victor(result[0], result[1])
 }
 
+export const buildGraph = (nodes) => {
+  const graph = new Graph()
+
+  for (let i = 0; i < nodes.length - 1; i++) {
+    const node1 = nodes[i]
+    const node2 = nodes[i + 1]
+    graph.addNode(node1)
+    graph.addNode(node2)
```

Comment:
> Does it matter that you're adding both nodes each time? You would have to handle the edge case if you didn't do it this way, but it would save almost half the work on large lists of nodes.

You're right. addNode handles it, but that's sloppy and I'll fix.